src/main/java/net/fabricmc/loader/api/entrypoint/EntrypointContainer.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2016 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.loader.api.entrypoint;
+
+import net.fabricmc.loader.api.ModContainer;
+
+public interface EntrypointContainer<T> {
+	T getEntrypoint();
+
+	ModContainer getProvidingModContainer();
```

Comment:
```java
	ModContainer getProvider();
```

Reply:
Yea I was definitely unsure on the naming for that one